src/ol_infrastructure/lib/aws/route53_helper.py

```diff
@@ -0,0 +1,25 @@
+import boto3
+import pulumi
+from typing import Text, Tuple
+
+route53_client = boto3.client('route53')
+
+
+def zone_opts(domain: Text) -> pulumi.ResourceOptions:
+    """Look up and conditionally import an existing hosted zone.
+
+    :param domain: The domain name to be looked up and optionally imported.  e.g. odl.mit.edu
+    :type domain: Text
+
+    :returns: A Pulumi ResourceOptions object that allows for importing unmanaged zones
+
+    :rtype: pulumi.ResourceOptions
+    """
+    zone = route53_client.list_hosted_zones_by_name(DNSName=domain)['HostedZones'][0]
+    zone_id = zone['Id'].split('/')[-1]  # 'Id' attribute is of the form /hostedzone/<ZONE_ID>
+    tags = route53_client.list_tags_for_resource(ResourceType='hostedzone', ResourceId=zone_id)
+    if 'pulumi_managed' in {tag['Key'] for tag in tags['ResourceTagSet']['Tags']}:  # noqa: C412
+        opts = pulumi.ResourceOptions()
+    else:
+        opts = pulumi.ResourceOptions(import_=zone_id)
```

Comment:
Is `import_` right?

Reply:
Yeah, that's so as to avoid overloading the builtin keyword.